systemtests/bin/download_sym_files.py

```diff
@@ -68,28 +156,46 @@ def download_sym_files(base_url, csv_file):
                 url = url + "?try"
             click.echo(click.style(f"Working on {url} ...", fg="yellow"))
 
-            # Download the file
-            headers = {"User-Agent": "tecken-systemtests"}
-            resp = requests.get(url, headers=headers, timeout=60)
+            headers = {
+                "User-Agent": "tecken-systemtests",
+                # We know storage backends will honor "Accept-Encoding": "gzip",
+                # so we test the unusual case only to ensure the response is
+                # still gzipped.
+                "Accept-Encoding": "identity",
+            }
+
+            for method in ["HEAD", "GET"]:
+                resp = requests.request(
+                    method,
+                    url,
+                    headers=headers,
+                    timeout=60,
+                )
 
-            for item in resp.history:
-                if item.status_code in (301, 302):
-                    click.echo(f">>> redirect: {item.url}")
-            click.echo(f">>> final: {resp.url}")
-            click.echo(f">>> status code: {resp.status_code}")
-            if resp.status_code == 200:
-                click.echo(f">>> file size {len(resp.content):,} bytes")
-
-            # Compare status code with expected status code
-            if resp.status_code != int(expected_status_code):
-                click.echo(
-                    click.style(
-                        f"FAIL: Status code: {resp.status_code} != {expected_status_code}",
-                        fg="red",
+                if resp.status_code != int(expected_status_code):
+                    click.echo(
+                        click.style(
+                            f"FAIL: {method} status code: {resp.status_code} != {expected_status_code}",
+                            fg="red",
+                        )
                     )
-                )
-            else:
-                click.echo(click.style("Success!", fg="green"))
+                else:
+                    click.echo(click.style(f"SUCCESS: {method} request!", fg="green"))
+
+                # If we're testing a 404, we should not expect the same response
+                # headers from the storage backend as for a 200 (e.g. Content-Encoding
+                # isn't included in a 404 response).
+                if method == "GET" and expected_status_code != "404" and test_headers:
```

Comment:
I'd go a step further and use `excpected_status_code == "200"`. Testing the storage backend headers for any other status code doesn't make sense.